### PR TITLE
Move partest out to its own repo: scala/scala-partest.

### PR DESCRIPTION
**Pull request by adriaanm**
As partest is now resolved from maven, `test/partest` uses `ant test.suite.init`
to determine the classpath (serialized to build/pack/partest.properties)
that's necessary to run `scala.tools.partest.nest.ConsoleRunner`.

Thus, partest gets exactly the same classpath, whether run from
the command line through `test/partest` or via `ant test`.

The version of partest we're using is specified by
properties defined in `starr.number`. Currently:

```
scala.binary.version=2.11.0-M4
partest.version.number=1.0-RC1
```

NOTES:
- The version of Scala being tested must be backwards binary compatible with
  the version of Scala that was used to compile partest.
- Need scalacheck on classpath for test/partest scalacheck tests.
- Remove atrophied ant tests (haven't been run/changed for at least two years
  I checked 81d659141a as a "random" sample).

review by everyone (@gkossakowski, @jamesiry, @jsuereth, @lrytz, @paulp, @retronym, @soc)

### CODE REVIEW

**Comment by adriaanm** [comment]
Here's the release of partest used as of this PR: https://github.com/scala/scala-partest/releases/tag/v1.0-RC1

**Comment by adriaanm** [comment]
(I'm mostly away from an internet connection until Monday, so apologies in advance for my delay in replying to your comments.)

**Comment by gkossakowski** [comment]
> The version of partest we're using is specified by properties defined in `starr.number`. Currently:
> 
> ```
> scala.binary.version=2.11.0-M4
> partest.version.number=1.0-RC1
> ```

Why partest version is stored in `starr.number` file?

**Comment by adriaanm** [comment]
> Why partest version is stored in starr.number file?

because this also specifies the version of the compiler used to compile the compiler
we should factor out more properties from build.xml and evolve starr.number to a more general properties file,
(so that you don't have to edit a huge xml file to change a version number -- for now I opted for a low-overhead solution that does not involve creating yet another small properties file)

**Comment by gkossakowski** [comment]
> because this also specifies the version of the compiler used to compile the compiler

Shouldn't partest be cross-versioned against STARR version then?

**Comment by adriaanm** [comment]
Yes, in principle, though this may be different (as long as they're binary compatible)

**Comment by adriaanm** [comment]
PLS REBUILD/pr-rangepos-per-commit@f01a673

**Comment by lrytz** [comment]
what do i need to do when i introduce binary incompatible changes to the compiler? i think there should be a smooth path for this scenario. otherwise we risk to get stuck with a compiler API that was never designed to be one.

**Comment by paulp** [comment]
>  otherwise we risk to get stuck with a compiler API that was never designed to be one.

See that speck on the horizon? That's the ship!

**Comment by gkossakowski** [comment]
@lrytz: Split your changes into two stages. The first stage introduces a new functionality and keeps the old version and deprecates it. Once a new milestone is released and partest switches to it you are free to modify partest to use new functionality and remove the old api from the compiler.

With cake pattern and it's (ab)use of inheritance this might be sometimes tricky but I think we can go rather far with this. I'm testing this strategy right now as I'm trying to refactor internals of the compiler so SymbolTable doesn't depend on rest of the cake (the whole Global) anymore. It's been rough ride in general and API compatibility requirement added a little bit more complexity but it wasn't a deal breaker so far.

**Comment by adriaanm** [comment]
So, I'd like to merge this -- could I get a verdict, @lrytz, and others?

**Comment by gkossakowski** [comment]
@adriaanm: this is not a blocker for merging this PR but I'd like to see follow up PR which cleans up the starr.number thing. It's really confusing the way it's done right now.

File: build.xml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<project name="sabbus" default="build" xmlns:artifact="urn:maven-artifact-ant">
+<project name="sabbus" default="build"
+  xmlns:artifact="urn:maven-artifact-ant"
+  xmlns:rsel="antlib:org.apache.tools.ant.types.resources.selectors">
```
Comment (gkossakowski) [inline build.xml:5]: Why this change? (I don't see any mention in commit message about this).

File: build.xml [hunk 2]
```diff
@@ -704,25 +705,49 @@ TODO:
...
+      <path refid="quick.interactive.build.path">
+      -->
+
+      <!-- <pathelement location="${build-pack.dir}/lib/scalap.jar"/>
+      TODO: move scalap out of repo
+       we don't test scalap as such (we *use* it for testing, though(
+       it's in partest.classpath, and not filtered out, so no need to depend on the one we built -->
+
+      <!-- include partest and its run-time dependencies, but filter out the compiler we just built,
+           as that's what we want to test! -->
```
Comment (gkossakowski) [inline build.xml:729]: Shouldn't partest dependencies on compiler be marked as "provided"?

See http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html for description of provided dependencies.

File: build.xml (hunk 1 above)
Comment (adriaanm) [inline build.xml:5]: It's for the `<restrict>` task

File: build.xml (hunk 2 above)
Comment (adriaanm) [inline build.xml:729]: Not necessarily, as we can also use it to test, e.g., scala-xml using the starr compiler. In this case, we don't want to do that, of course.

**Comment by adriaanm** [comment]
What should we rename it to? `dependencies.properties`?

**Comment by adriaanm** [comment]
PLS REBUILD ALL

(Seeing if all still passes when merged into current master.)
